### PR TITLE
Implement comparing outputs that are sequence of maps of strings to floats

### PR DESCRIPTION
**Pull request by ryanlai2**
This change implements 2 things for winml model tests:

- Moves the creation of Ort::Value from ITensor into ort_value_helpers.cpp
- Implement the comparing of outputs that are of type sequence of maps of strings to floats

### CODE REVIEW

File: winml/test/model/compare_feature_value.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+namespace CompareFeatureValuesHelper{
+
+template <typename T>
+bool IsResultCloselyMatch(const T& outvalue, const T& expected_value, const double diff, const double tol) {
+  if (diff > tol) return false;
+  if (std::isnan(diff) && !(std::isnan(outvalue) && std::isnan(expected_value)) &&
+      !(std::isinf(outvalue) && std::isinf(expected_value)))
+    return false;
+  return true;
+}
```
Comment (tiagoshibata) [inline winml/test/model/compare_feature_value.cpp:19]: Should diff and tol be of type T?

**Review by tiagoshibata — CHANGES_REQUESTED** [review]

File: winml/test/model/compare_feature_value.cpp (hunk 1 above)
Comment (ryanlai2) [reply winml/test/model/compare_feature_value.cpp:19]: I followed this example here : https://github.com/microsoft/onnxruntime/blob/53307a5f2ea5a665a80fc55b40965c95f70bd1f6/onnxruntime/test/util/compare_ortvalue.cc#L59

perSampleTolerance and relativePerSampleTolerance are both always type double so I think it should continue to carry that datatype in calculations.

File: winml/test/model/ort_value_helper.cpp [hunk 2]
```diff
@@ -136,4 +136,80 @@ winml::ITensor LoadTensorFromOrtValue(Ort::Value& val) {
...
+            shape.data(),
+            shape.size(),
+            ONNXTensorElementDataType::ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING));
+    std::vector<const char*> strData;
+    auto strValues = winmlTensor.as<TensorString>().GetAsVectorView();
+    for (winrt::hstring str : strValues) {
+      std::string utf8Str = _winml::Strings::UTF8FromHString(str);
+      strData.push_back(utf8Str.c_str());
+    }
+    WINML_EXPECT_NO_THROW(ortValueCreated.FillStringTensor(strData.data(), strData.size()));
```
[the remark this answers is not in the document]
Comment (ryanlai2) [reply winml/test/model/ort_value_helper.cpp]: Good catch! Thanks, I think I've fixed it.

File: winml/test/model/ort_value_helper.cpp [hunk 3]
```diff
@@ -136,4 +136,81 @@ winml::ITensor LoadTensorFromOrtValue(Ort::Value& val) {
...
+        ortValueCreated = Ort::Value::CreateTensor(
+            allocator,
+            shape.data(),
+            shape.size(),
+            ONNXTensorElementDataType::ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING));
+    std::vector<const char*> strData;
+    std::vector<std::string> utf8Strs;
+    auto strValues = winmlTensor.as<TensorString>().GetAsVectorView();
+    for (winrt::hstring str : strValues) {
+      utf8Strs.push_back(std::move(_winml::Strings::UTF8FromHString(str)));
```
Comment (tiagoshibata) [inline winml/test/model/ort_value_helper.cpp:209]: Nit: It's a pessimising move that will trigger a warning in GCC and clang depending on the warning flags used (https://developers.redhat.com/blog/2019/04/12/understanding-when-not-to-stdmove-in-c/)

**Review by tiagoshibata — APPROVED** [review]

**Pull request merged**